### PR TITLE
feat: Add math and algo book to contest table (#2785)

### LINKED ISSUE #2785: [Feat] 数学・アルゴ本の問題をインポートできるようにしましょう

### PR DESCRIPTION
**Pull request by KATO-Hiro**
close #2785

See #662 

## Summary by CodeRabbit

* **New Features**
  * Added a Math & Algorithm contest group with custom display behavior and provider support.
  * Expanded problem catalog with 20+ new contest tasks from ABC, ARC, Typical90, Panasonic, JSC, DP, and others.

* **Documentation**
  * Added a step-by-step guide for importing contest-task CSV data into the database, including preparation, import steps, troubleshooting, and lessons learned.

* **Tests**
  * Added unit tests and mock data covering the Math & Algorithm provider (filtering, metadata, table generation, sorting, and edge cases).

### CODE REVIEW

File: prisma/tasks.ts [hunk 1]
```diff
@@ -4531,6 +4622,13 @@ export const tasks = [
     title: 'D. Knapsack 1',
     grade: 'Q3',
   },
+  {
+    id: 'dp_d',
+    contest_id: 'dp',
+    problem_index: 'D',
+    name: 'Knapsack 1',
+    title: 'D. Knapsack 1',
+  },
```
[the remark this answers is not in the document]
Comment (KATO-Hiro) [reply prisma/tasks.ts]: ありがとう．修正します
